composer.json

```diff
@@ -41,5 +41,8 @@
             "dealerdirect/phpcodesniffer-composer-installer": true,
             "phpstan/extension-installer": true
         }
+    },
+    "require-dev": {
+        "phpstan/phpstan-strict-rules": "^1.1"
```

Comment:
> Installed with a downstream repo to ensure it does not cause unintended issues. Worked as expected.

I don't see `phpstan/phpstan-strict-rules` in any of our other repos.

Are we adding a new ruleset that will be automatically included and enforced by phpstan in any repo that includes this latest version of `usher`?

Reply:
No, exactly the opposite. By putting this in dev dependencies it will ensure that it is _not_ included in downstream repos.

Reply:
Ah, I didn't think about that feature/angle of using dev dependencies before. The more you know.

Reply:
I am so confused. I think you both might be saying the same thing? 

My primitive (non)understanding is that by putting it into require-dev, it won't be automatically included downstream, but is _available_ downstream for a dev install, and isn't that what @adamzimmermann is saying?

🤔

Reply:
> but is available downstream for a dev install, and isn't that what @adamzimmermann is saying?

I wasn't saying that, but that is a good point. I didn't know the answer to this, so I did some research.

> NB composer install will only download and install the development dependencies of your application or package and not the development dependencies of dependencies. So if you have a dependency on "Package A" and "Package A" has a development dependency on "Package B" only "Package A" will be downloaded and installed in your project.

— https://dev.to/timoschinkel/good-practices-when-working-with-composer-5a1c

This also lines up with what @markdorison is implying, so that's good.